src/app/admin/members/table.tsx

```diff
@@ -65,24 +53,51 @@ export function MembersAdminTable() {
                         width: 80,
                         textAlign: 'center',
                         title: <Box mr={6}>Edit</Box>,
-                        render: (member) => (
-                            <Group gap={4} justify="center" wrap="nowrap">
-                                <ActionIcon size="sm" variant="subtle" color="blue" onClick={() => setEditing(member)}>
-                                    <IconEdit size={18} />
-                                </ActionIcon>
-                                <SuretyGuard onConfirmed={() => deleteMember(member.identifier)}>
-                                    <IconTrash size={18} />
-                                </SuretyGuard>
-                            </Group>
-                        ),
+                        render: (member) => <MemberRow member={member} />,
                     },
                 ]}
             />
 
-            <EditModal editing={editing} onComplete={onEditComplete} />
-            <Flex justify={'end'} mt="lg">
-                <Button onClick={() => setEditing(NEW_MEMBER)}>Add new Member</Button>
-            </Flex>
+            <AddMember />
+        </Flex>
+    )
+}
+
+const AddMember: FC = () => {
+    const [opened, { open, close }] = useDisclosure(false)
+
+    return (
+        <Flex justify={'end'} mt="lg">
+            <Modal opened={opened} onClose={close} title={`Add Member`} closeOnClickOutside={false}>
+                <EditMemberForm member={NEW_MEMBER} onCompleteAction={close} />
+            </Modal>
+            <Button onClick={open}>Add new Member</Button>
         </Flex>
     )
 }
+
+const MemberRow: FC<{ member: Member }> = ({ member }) => {
+    const queryClient = useQueryClient()
+    const [opened, { open, close }] = useDisclosure(false)
+
+    const { mutate: deleteMember } = useMutation({
+        mutationFn: deleteMemberAction,
+        onSettled: async () => {
+            return await queryClient.invalidateQueries({ queryKey: ['members'] })
+        },
+    })
+
+    return (
+        <Group gap={4} justify="center" wrap="nowrap">
+            <Modal opened={opened} onClose={close} title={`Edit ${member.name}`} closeOnClickOutside={false}>
+                <EditMemberForm member={member} onCompleteAction={close} />
```

Comment:
I laid the groundwork here to potentially hoist the need for passing down an `onCompleteAction`, previously it was being passed down through 3-4 different components just to close the modal when the form was submitted. Each form should have it's own modal now which eliminates another layer, i still think this can be improved on though 🤔 can come back to this down the road